crypto/s2n_rsa_pss.c

```diff
@@ -112,20 +107,26 @@ static int s2n_rsa_pss_validate_sign_verify_match(const struct s2n_pkey *pub, co
     return 0;
 }
 
-static int s2n_rsa_validate_param_equal(const RSA *pub, const RSA *priv, int required,
-                                        ossl_get_rsa_param_fn get_rsa_param_fn)
+static int s2n_rsa_validate_params_equal(const RSA *pub, const RSA *priv)
 {
-    const BIGNUM *pub_val = get_rsa_param_fn(pub);
-    const BIGNUM *priv_val = get_rsa_param_fn(priv);
+    const BIGNUM *pub_val_e = NULL;
+    const BIGNUM *pub_val_n = NULL;
+    RSA_get0_key(pub, &pub_val_n, NULL, &pub_val_e);
+
+    const BIGNUM *priv_val_e = NULL;
+    const BIGNUM *priv_val_n = NULL;
+    RSA_get0_key(priv, &priv_val_n, NULL, &priv_val_e);
 
-    if (required && (pub_val == NULL || priv_val == NULL)) {
+    if (pub_val_e == NULL && priv_val_e == NULL) {
```

Comment:
pardon my ignorance, but should this be an `||`?

Reply:
Yes it should be. I mistakenly swapped the order of `d` and `e` when calling [RSA_get0_key](https://www.openssl.org/docs/manmaster/man3/RSA_get0_key.html) and will fix that as well.

Reply:
hmm.. why didn't tests fail?

Reply:
I believe that tests didn't fail in this case because this case would never evaluate to `true`. This is because both the public and private key should have the public exponent `e`, and the private key would have the private exponent `d`. 

Since this checked if both were NULL, it would not result in an error. Had I used `||` originally, this would not have passed tests.